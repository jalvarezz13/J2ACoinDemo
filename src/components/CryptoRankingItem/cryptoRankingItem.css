/* General styles */
.item-container p {
  font-family: Helvetica, Arial, sans-serif;
  font-size: 14px;
}

.item-container {
  width: 100%;

  display: grid;
  grid-template-columns: 1.5fr 2.5fr 4.5fr 4fr 3fr 6fr 6fr;
  align-content: center;
  justify-content: space-between;
  align-items: center;
  justify-items: stretch;

  padding: 10px 0px;
  border-bottom: 1px solid white;
}

/* Styles for prices and changes in the price */
.change-value-positive {
  color: #16c784;
}

.change-value-negative {
  color: #ea3943;
}

.change-value-positive::before,
.change-value-negative::before {
  font-size: 12px;
  vertical-align: text-top;
}

.change-value-positive::before {
  content: "▲ ";
}

.change-value-negative::before {
  content: "▼ ";
}

/* Styles for the graph of 7d */

.item-change-7d-text {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-around;
}

.item-change-7d {
  display: flex;
  flex-direction: column;
}

.graph-7d-negative,
.graph-7d-positive {
  width: 100%;
  height: 50px;
  text-align: center;
  box-sizing: content-box;
  padding-top: 2px;
  padding-bottom: 3px;
}

.graph-7d-positive {
  filter: hue-rotate(85deg) saturate(80%) brightness(0.85);
}

.graph-7d-negative {
  filter: hue-rotate(300deg) saturate(210%) brightness(0.7) contrast(170%);
}

/* Coin logo */
.item-logo {
  width: 50px;
  -webkit-filter: drop-shadow(3px 3px 2px rgba(0, 0, 0, 0.7));
  filter: drop-shadow(3px 3px 2px rgba(0, 0, 0, 0.7));
}

/* Item id */
.item-id {
  text-align: center;
}

/* Item price */
.price-container {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  align-content: center;
  justify-content: flex-start;
  align-items: center;
}

.item-price {
  font-size: 15px;
  text-align: left;
  font-weight: bold;
}

/* Proof of ... icon */
.proof-of {
  width: 25px;
  padding-left: 5px;
}

/* Item name */
.item-name {
  text-align: left;
}

/* MODO MOVIL */
@media (max-width: 950px) {
  .item-container {
    width: 100%;
  
    display: flex;
    flex-direction: column;
  }
}