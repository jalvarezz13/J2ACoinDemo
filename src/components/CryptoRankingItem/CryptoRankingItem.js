import React, { useState } from "react"
import PropTypes from "prop-types"
import strongbox from "../../assets/strongbox.svg"
import pick from "../../assets/pick.svg"
import "./cryptoRankingItem.css"

const CryptoRankingItem = ({ item }) => {
  const [itemData] = useState(item)
  const proofOf =
    (itemData.tags.includes("pow") && <img className="proof-of" src={pick} alt="PoW"></img>) ||
    (itemData.tags.includes("pos") && <img className="proof-of" src={strongbox} alt="PoS"></img>)

  return (
    <div className="item-container">
      <p className="item-id">{itemData.cmc_rank} </p>
      <img className="item-logo" src={`https://s2.coinmarketcap.com/static/img/coins/64x64/${itemData.id}.png`} alt={`Logo de ${itemData.name}`} />
      <p className="item-name">
        {itemData.name} ({itemData.symbol})
      </p>
      <div className="price-container">
        <span className="item-price">{itemData.quote.USD.price.toFixed(4)} $</span>
        {proofOf}
      </div>
      <p id="change_24" className={itemData.quote.USD.percent_change_24h > 0 ? "change-value-positive" : "change-value-negative"}>
        {parseFloat(itemData.quote.USD.percent_change_24h).toFixed(4)}%
      </p>
      <p>{new Intl.NumberFormat("de-DE").format(itemData.quote.USD.market_cap.toFixed(2))}$</p>
      <div className="item-change-7d">
        <div className="item-change-7d-text">
          <p>7 días</p>
          <p className={itemData.quote.USD.percent_change_7d > 0 ? "change-value-positive" : "change-value-negative"}>
            {parseFloat(itemData.quote.USD.percent_change_7d).toFixed(4)}%
          </p>
        </div>
        <img
          src={`https://s3.coinmarketcap.com/generated/sparklines/web/7d/2781/${itemData.id}.svg`}
          alt={`Gráfico a 7 días de ${itemData.name}`}
          className={itemData.quote.USD.percent_change_7d > 0 ? "graph-7d-positive" : "graph-7d-negative"}
        />
      </div>
    </div>
  )
}

CryptoRankingItem.propTypes = {
  item: PropTypes.object.isRequired
}

export default CryptoRankingItem
