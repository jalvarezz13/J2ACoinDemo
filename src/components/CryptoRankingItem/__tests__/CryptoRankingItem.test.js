import React from "react"
import { shallow } from "enzyme"
import CryptoRankingItem from "../CryptoRankingItem"

const rightJson = {
  circulating_supply: 18994618,
  cmc_rank: 1,
  date_added: "2013-04-28T00:00:00.000Z",
  id: 1,
  last_updated: "2022-03-26T18:20:00.000Z",
  max_supply: 21000000,
  name: "Bitcoin",
  num_market_pairs: 9274,
  platform: null,
  quote: {
    USD: {
      fully_diluted_market_cap: 931966275464.83,
      last_updated: "2022-03-26T18:20:00.000Z",
      market_cap: 842968732920.8187,
      market_cap_dominance: 42.0811,
      percent_change_1h: 0.03335492,
      percent_change_24h: 0.04590512,
      percent_change_30d: 23.34775013,
      percent_change_60d: 18.90127295,
      percent_change_7d: 5.83178147,
      percent_change_90d: -11.66697751,
      price: 44379.34645070613,
      volume_24h: 16515063157.110834,
      volume_change_24h: -46.1275,
    },
  },
  self_reported_circulating_supply: null,
  self_reported_market_cap: null,
  slug: "bitcoin",
  symbol: "BTC",
  tags: [
    "mineable",
    "pow",
    "sha-256",
    "store-of-value",
    "state-channel",
    "coinbase-ventures-portfolio",
    "three-arrows-capital-portfolio",
    "polychain-capital-portfolio",
    "binance-labs-portfolio",
    "blockchain-capital-portfolio",
    "boostvc-portfolio",
    "cms-holdings-portfolio",
    "dcg-portfolio",
    "dragonfly-capital-portfolio",
    "electric-capital-portfolio",
    "fabric-ventures-portfolio",
    "framework-ventures-portfolio",
    "galaxy-digital-portfolio",
    "huobi-capital-portfolio",
    "alameda-research-portfolio",
    "a16z-portfolio",
    "1confirmation-portfolio",
    "winklevoss-capital-portfolio",
    "usv-portfolio",
    "placeholder-ventures-portfolio",
    "pantera-capital-portfolio",
    "multicoin-capital-portfolio",
    "paradigm-portfolio",
  ],
  total_supply: 18994618,
}
const wrongJson = {
  circulating_supply: 18994618,
  cmc_rank: 1,
  date_added: "2013-04-28T00:00:00.000Z",
  id: 1,
  last_updated: "2022-03-26T18:20:00.000Z",
  max_supply: 21000000,
  name: "Bitcoin_malo",
  num_market_pairs: 9274,
  platform: null,
  quote: {
    USD: {
      fully_diluted_market_cap: 931966275464.83,
      last_updated: "2022-03-26T18:20:00.000Z",
      market_cap: 842968732920.8187,
      market_cap_dominance: 42.0811,
      percent_change_1h: 0.03335492,
      percent_change_24h: -0.04590512,
      percent_change_30d: 23.34775013,
      percent_change_60d: 18.90127295,
      percent_change_7d: 5.83178147,
      percent_change_90d: -11.66697751,
      price: 44389.34645070613,
      volume_24h: 16515063157.110834,
      volume_change_24h: -46.1275,
    },
  },
  self_reported_circulating_supply: null,
  self_reported_market_cap: null,
  slug: "bitcoin",
  symbol: "BTC",
  tags: [
    "mineable",
    "pow",
    "sha-256",
    "store-of-value",
    "state-channel",
    "coinbase-ventures-portfolio",
    "three-arrows-capital-portfolio",
    "polychain-capital-portfolio",
    "binance-labs-portfolio",
    "blockchain-capital-portfolio",
    "boostvc-portfolio",
    "cms-holdings-portfolio",
    "dcg-portfolio",
    "dragonfly-capital-portfolio",
    "electric-capital-portfolio",
    "fabric-ventures-portfolio",
    "framework-ventures-portfolio",
    "galaxy-digital-portfolio",
    "huobi-capital-portfolio",
    "alameda-research-portfolio",
    "a16z-portfolio",
    "1confirmation-portfolio",
    "winklevoss-capital-portfolio",
    "usv-portfolio",
    "placeholder-ventures-portfolio",
    "pantera-capital-portfolio",
    "multicoin-capital-portfolio",
    "paradigm-portfolio",
  ],
  total_supply: 18994618,
}

describe("CryptoRankingItem tests", () => {
  test("Should <CryptoRankingItem /> renders correctly", () => {
    const wrapper = shallow(<CryptoRankingItem item={rightJson} />)
    expect(wrapper).toMatchSnapshot()
  })

  test("Should show an incorrect price", () => {
    const wrapper2 = shallow(<CryptoRankingItem item={wrongJson} />)
    const price = wrapper2.find("span").text().trim()
    expect(price).toBe("44389.3465 $")
  })

  test("Should show the correct class for a negative change", () => {
    const wrapper3 = shallow(<CryptoRankingItem item={wrongJson} />)
    const change_24_negative = wrapper3.find("#change_24").props()["className"]
    expect(change_24_negative).toBe("change-value-negative")
  })

  test("Should show the correct class for a positive change", () => {
    const wrapper4 = shallow(<CryptoRankingItem item={rightJson} />)
    const change_24_positive = wrapper4.find("#change_24").props()["className"]
    expect(change_24_positive).toBe("change-value-positive")
  })
})
