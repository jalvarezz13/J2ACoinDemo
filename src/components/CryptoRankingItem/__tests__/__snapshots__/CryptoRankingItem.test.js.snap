// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CryptoRankingItem tests Should <CryptoRankingItem /> renders correctly 1`] = `ShallowWrapper {}`;
